source/common/router/rds_impl.cc

```diff
@@ -75,6 +75,8 @@ RdsRouteConfigProviderImpl::RdsRouteConfigProviderImpl(
   // then there is no actual RDS server, and hence no RDS cluster name.
   if (rds.has_config_source() && rds.config_source().has_api_config_source()) {
     cluster_name_ = rds.config_source().api_config_source().cluster_names()[0];
+  } else if (rds.has_config_source() && rds.config_source().has_ads()) {
+    cluster_name_ = "ADS_CLUSTER";
```

Comment:
I tried to get the actual ads cluster name, but looks like it requires a bigger change, like adding ads cluster name while building the `ClusterManager` and have to introduce a public method in `ClusterManager` which may be overkill for this. Please let me know if you would like me to do it or if there is  a simpler way to get the actual ads cluster name.

Reply:
This is actually pretty tricky for a few reasons:
1. We're migrating `envoy.api.v2.ApiConfigSource` (by which ADS) is configured away from using raw cluster names to `envoy.api.v2.GrpcService`.
2. The ADS `GrpcService` might either use a cluster name (when using Envoy gRPC) or a raw address (when using Google gRPC). There is a `stat_prefix` in the latter case, which could be used for reporting, but I think this would be confusing semantics here.

I think including cluster name in the admin console is not the way to go long term. What we should do perhaps is include a JSON serialization of `rds.config_source().api_config_source()`, or elide it completely. 

@junr03 can probably comment further on the original intention of this field and what it might be used for in a post-cluster centric view of gRPC endpoints.

Reply:
Will defer to @junr03 on the final solution here, but in general I think we need to move to a model where we just store the config in proto form and are able to spit it back out. See also https://github.com/envoyproxy/envoy/issues/2421

Reply:
The original purpose of that field was to be able to give a sense to someone operating the admin interface where the RDS config came from, given that the configuration might have come from an LDS response. 

In general,  #2421 and #2172 show that there is a need to be able to understand what is currently loaded in envoy and where it came from. There are a couple segmented pieces here and there but a comprehensive solution would be great.

As far as this particular change goes, I am ok with it in order to make the output of the /routes admin endpoint correct and more clear for now.

Reply:
What about just dumping the full `rds.config_source.api_config_source()` proto instead of cluster name? It seems this would kill multiple birds with one stone.

Reply:
Sorry, in my comment above I meant `it` to mean your suggestion of dumping `rds.config_source().api_config_source()` -- I elided a sentence. However, I would do `rds.config_source()` so all three sources are covered by default?

Reply:
SGTM